packages/shared-ui/src/flowHandler/flowHandler.ts

```diff
@@ -35,37 +35,33 @@ export class FlowHandler {
    * The constructor initializes the FlowHandler with a flow name, a project configuration, and a flow handler configuration.
    * It sets the current flow to the specified flow, the current screen to the Start screen, and initializes the screen history as an empty array.
    */
-  constructor(
-    corbadoApp: CorbadoApp | undefined,
-    projectConfig: ProjectConfig,
-    onLoggedIn: () => void,
-    initialFlowType?: FlowType,
-  ) {
+  constructor(corbadoApp: CorbadoApp | undefined) {
     if (!corbadoApp) {
       throw new Error('corbadoApp is undefined. This should not happen.');
     }
 
+    this.#corbadoApp = corbadoApp;
+    this.#screenHistory = [];
+  }
+
+  /**
+   * Initializes the FlowHandler.
+   * Call this function after registering all callbacks.
+   */
+  async init(i18next: i18n, projectConfig: ProjectConfig, onLoggedIn: () => void, initialFlowType?: FlowType) {
```

Comment:
We should now remove the `projectConfig` params separately. Initially, it was in a constructor since we were unable to make async calls. Now, we should use `corbadoApp.projectService.getProjectConfig()` since it has a cache mechanism and we can avoid two project config calls this way. Also, that would make sure that the CorbadoApp and ProjectConfig belongs to same project.